client/my-sites/customer-home/controller.jsx

```diff
@@ -28,9 +28,7 @@ export default async function renderHome( context, next ) {
 		window.scrollTo( 0, 0 );
 	}
 
-	context.primary = (
-		<CustomerHome key={ site.ID } showLaunchpadFirst={ shouldShowLaunchpadFirst( site ) } />
-	);
+	context.primary = <CustomerHome site={ site } />;
```

Comment:
Unless there's a reason we don't need to re-instantiate the `CustomerHome` when the site changes anymore, then we should keep the `key` prop on this element.

Re-rendering when the site prop changes isn't the same thing as [changing the key when you want to reset _everything_](https://react.dev/learn/preserving-and-resetting-state#option-2-resetting-state-with-a-key).

```suggestion
	context.primary = <CustomerHome key={ site.ID } site={ site } />;
```

Reply:
Good catch, thanks!